### ISSUE TITLE
NetworkChange.NetworkAddressChanged bug

### ISSUE THREAD

**Issue by Bloodmag** [dotnet/corefx#41203]
# Issue Title

NetworkAddressChanged calls a callback just once.

# General

I wrote a simple program to test how NetworkChange.NetworkAddressChanged and got some really weird results: if I subscribe this event, it's being called just once, but if I add and remove a subscribtion to an empty function, it starts working just fine (one call for plugging ethernet cable out and two for plugging in).

It works okay in .Net framework.

I've tried changing event causes, for example, changing my computer's ip on DHCP server and ipconfig /renew. Target platform is <TargetFramework>netcoreapp2.2</TargetFramework>.
OS: Windows 10 pro

Code to reproduce the issue:

```
class Program
{
    static void Main(string[] args)
    {
        //Subscribing event
        NetworkChange.NetworkAddressChanged += OnAddressChanged;
        //just a loop
        while (true)
        {
            Thread.Sleep(1000);
        }
    }
    //Callback
    static void OnAddressChanged(object o, EventArgs e)
    {
        Console.WriteLine(e.ToString());

        foreach (NetworkInterface n in NetworkInterface.GetAllNetworkInterfaces())
        {
            Console.WriteLine("   {0} is {1}", n.Name, n.OperationalStatus);
        }
        //Works if uncomment two lines below
        //NetworkChange.NetworkAddressChanged += DuctTape;
        //NetworkChange.NetworkAddressChanged -= DuctTape;
    }
    //Empty function
    static void DuctTape(object sender, EventArgs e)
    {
        ;
    }
}
```

**Comment by nechvatalp** [comment]
I tested this and was able to reproduce it but only on 2.2.
.NET Core 3.0 seems to work as expected.

![capture](https://user-images.githubusercontent.com/38288693/65905783-7f925300-e3c1-11e9-97a6-506bfc097c26.png)

**Comment by nechvatalp** [comment]
Seems like this was fixed in #32945

**Comment by davidsh** [comment]
Thanks for verifying it was fixed in .NET Core 3.0. There are no plans to backport this fix to .NET Core 2.x.
